config.go

```diff
@@ -332,6 +332,13 @@ func loadConfig() (*config, error) {
 	// path to all of the files and directories that will live within it.
 	lndDir := cleanAndExpandPath(preCfg.LndDir)
 	if lndDir != defaultLndDir {
+		preCfg2 := config{} // Parse again the command line to check if a config file is set
+		if _, err := flags.Parse(&preCfg2); err != nil {
+			return nil, err
+		}
+		if preCfg2.ConfigFile == "" { // That means that the user didn't specify a config file
```

Comment:
Due to the comment above, we'll want to check if the config file path is the _default_, since that's what it'll be set to if it was not provided in the command line flags. If it was not provided, then it'll be the default and we should assume that the new config file lives under the specified `lnddir`. Otherwise, we'll leave it as is since the user intended the specified config file to override the one within `lnddir`.